banwebScrape.py

```diff
@@ -20,8 +24,40 @@
             MTU_subject,MTU_number,MTU_credits
 """
 
-# Fun stats information
-# total_bytes_transfered = 0
+
+def form_cli_args():
+    """
+    Constructs and validates CLI arguments
+
+    :returns: ArgParse object of all arguments
+    """
+    # Args
+    parser = argparse.ArgumentParser()
+    # logging
+    log_level = parser.add_mutually_exclusive_group()
+    log_level.add_argument("--debug", action="store_true",
+                           help="Toggles debug mode on")
+    log_level.add_argument("--log-level", default="INFO",
+                           choices=_nameToLevel.keys(),
+                           help="Toggles logger level")
+    parser.add_argument("--log-file", default=None, help="File to log to")
+    # Contents output
+    parser.add_argument("--output", default="transfer-info",
+                        help="Output file for the transfer data")
+    parser.add_argument("--output-type", default="csv",
+                        choices=Output_Types.get_names(),
+                        help="Filetype to save data in")
+    parser.add_argument("--minify", action="store_true",
+                        help="Minify's the file contents " +
+                        "(only works with JSON output)")
+    # Read in options
+    parser.add_argument("--in-file", default=None,
```

Comment:
I'll do a more thorough review (and test it locally) tomorrow, but passing an argument to `--in-file` just converts an existing file to a different format, correct? Might be worth making that more clear in the description.